infrastructure/code/code_test.go

```diff
@@ -394,16 +394,14 @@ func Test_LoadIgnorePatternsAndCountFiles_RelativePathIgnores(t *testing.T) {
 	subDir := filepath.Join(tempDir, "evilfolder")
 	_ = os.Mkdir(subDir, 0755)
 	writeGitIgnoreIntoDir("*", t, subDir)
-	expectedSubDirPattern, err := filepath.Rel(tempDir, filepath.Join(subDir, "**/*"))
-	assert.NoError(t, err)
-	expectedSubDirPattern = filepath.ToSlash(expectedSubDirPattern)
+	expectedSubDirPattern := filepath.ToSlash(filepath.Join(subDir, "**/*"))
 
 	sc := Scanner{}
-	_, err = sc.loadIgnorePatternsAndCountFiles(tempDir)
+	_, err := sc.loadIgnorePatternsAndCountFiles(tempDir)
 
 	assert.NoError(t, err)
 	assert.Contains(t, sc.ignorePatterns, expectedSubDirPattern)
-	assert.Len(t, sc.ignorePatterns, len(getDefaultIgnorePatterns())+1)
+	assert.Len(t, sc.ignorePatterns, len(getDefaultIgnorePatterns())+2)
```

Comment:
Why is it +2 now?

Reply:
Because the parsing into globs can return more than one glob - and it now does in this case.